### ISSUE TITLE
The following file doesn't recreate the curves upon opening. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#5685]
I think this is probably related to #5482

The following file doesn't recreate the curves upon opening. 

[curved slab.ifc.txt](https://github.com/user-attachments/files/17611986/curved.slab.ifc.txt)

https://github.com/user-attachments/assets/ab80a6a1-cffb-4047-9078-2addfab765a1

**Comment by sboddy** [comment]
Still an issue with:
```
blender_version: 4.4.1
bonsai_version: 0.8.3-alpha250420
```

**Comment by sboddy** [comment]
It's kind of funny that this is broken in two ways. You get too few segments, or you get too many. Never the Goldilocks number.

**Comment by sboddy** [comment]
I'm going to reopen this issue as I don't consider it complete after the two PR's #8368 (addr. #8051) and #8484 (addr. this) are applied. This is the result of loading of the test file attached to this ticket with the default values for Deflection Tol. and Angular Tol.:

<img width="1104" height="625" alt="Image" src="https://github.com/user-attachments/assets/83686646-c361-4563-8c2f-c0603f56487b" />

The lower curved slab from this original ticket is "not great" in my opinion. The added top piece is supposed to be a Circle Hollow Profile. The current algorithm degenerates rapidly for small sizes to four sides, and can even result in ugly intersecting geometry.

I've got a fix already. But maybe it is desirable to have really low grade shapes for some of the more extreme models. I just know I have pipes in my model that look like the one above, and I'd rather not.

The weird thing is that the fix I've got uses Angular Tolerance as well to set the "floor" which at the default of 0.5 rads is equiv. to 13 segments for a full circle. This setting does not appear to be used _anywhere_ else in the codebase. So it's not clear why it is even there in the first place. With this change I get:

<img width="1049" height="548" alt="Image" src="https://github.com/user-attachments/assets/9c470bc1-cce1-4f24-a2b1-2db29595c05a" />

So maybe we still want to revisit the Deflection Tolerance to address the low resolution of the lower slab, because for me it is too high at 0.05. I'd say 0.01 is the minimum I would say is appropriate:

<img width="1010" height="512" alt="Image" src="https://github.com/user-attachments/assets/763c952e-2d14-41ca-bd72-bf6e574291cc" />

Any thoughts @Moult (sorry for abusing your notification filtering again) @theoryshaw ?

**Comment by sboddy** [comment]
I've pushed it as a draft in anticipation of a decision on whether we want such low res. small stuff, and also whether the current deflection tolerance is low enough.

**Comment by sboddy** [comment]
@Moult Would you like some ketchup for those?

https://github.com/IfcOpenShell/IfcOpenShell/commit/247f0c41d96ff59d58edfbdf21693417652a3e10

# 🤭

**Comment by Moult** [comment]
#8484 is a fix to merely use the same settings in two spots, so it is a valid fix, but not expected to actually fix this.

#8368 is the real fix, but 1) it is a C++ change, so it only affects you if you use a build with it included and 2) it relies on `circle-segments` being set in Bonsai, which I didn't see code for in that PR (haven't grepped locally, so could be wrong). Can you confirm both? I see in your draft PR there is additional considerations so it's not just one or the other as @aothms suggested but it's a bit more conditional on the actual size (which intuitively makes sense but I haven't dug into it).

**Comment by sboddy** [comment]
@Moult Yeah, it get's confusing due to BIMvoice and I addressing the same issue initially. My investigation found two issues, one the same as BIMvoice, and one BIMvoice didn't see as he was addressing a different issue. BIMvoice merged first, so only the secondary part of my fix got merged.

It definitely needs to do something different from the current situation. @falken10vdl now also has a PR for forcing circle segments to 24, though I think he's forcing it in pure Python from a _very_ brief look at the commits.

I think for sure that 0.05 on the deflection tolerance is too low in general. If the previous 0.001 is too low for performance reasons, then maybe 0.01 is a reasonable compromise. the circular profiles getting collapsed into 4 sided squares is ugly to me, but some guy doing MEP for a chemical site with thousands of pipes might think that's a reasonable trade off. I'm not really the guy to make the final call on what the defaults are. I think my PR is a good compromise, if it is OK to bump default deflection tolerance up to 0.1, but then I'm biased. Then we have a reasonable floor no matter what size the objects.

**Comment by falken10vdl** [comment]
Yes I created a simple python based hack so for circles we get a number of segments (as property to be defined similar to deflection tolerance) https://github.com/IfcOpenShell/IfcOpenShell/pull/8854 but I think that a solution like @sboddy is more general since I guess It cover all circles segment cases? I just changed the PR to draft and will remove it.
Cheers!

**Comment by Moult** [comment]
I would prefer to defer to @aothms to make a call on this - he lives and breathes geometry.
